src/components/VenueBody/index.jsx

```diff
@@ -0,0 +1,120 @@
+import { PageWidthContainer } from "../../shared/styledComponents/pageWidthContainer.js";
+import { CircularProgress, Link, useMediaQuery, useTheme } from "@mui/material";
+import {
+  StyledArrowBackIosIcon,
+  StyledBackToResultsFlexDiv,
+  StyledBackToResultsLinkContainer,
+  StyledBodyContainer,
+  StyledLeftColumnContainer,
+  StyledRightColumnContainer,
+  StyledWideBodyContainer,
+} from "./Venue.styled.js";
+import { DetailsAndImageContainer } from "./DetailsAndImageContainer/index.jsx";
+import React, { useRef } from "react";
+import { useVenue } from "./useVenue.js";
+import { WideBodyLinkBarAndContentContainer } from "./LinkBarAndBody/WideBodyLinkBarAndContentContainer.jsx";
+import { VerticalContainer } from "../../shared/styledComponents/verticalContainer.styled.js";
+import { ContactInfo } from "./ContactInfo/index.jsx";
+import DatePicker from "./DatePicker/index.jsx";
+import { NarrowBodyLinkBar } from "./LinkBarAndBody/NarowBodyLinkBar.jsx";
+import { Description } from "./Description/index.jsx";
+import { Gallery } from "./Gallery/index.jsx";
+import MapWithAddress from "./MapWithAddress/index.jsx";
+import { useLinkBar } from "./LinkBarAndBody/useLinkBar.js";
+import { BookDrawer } from "./BookDrawer.jsx";
+
+const DisplayedContentValue = {
+  description: "description",
+  gallery: "gallery",
+  map: "map",
+  contacts: "contacts",
+};
+
+export function Venue({ venueId }) {
+  const descriptionRef = useRef(null);
+  const galleryRef = useRef(null);
+  const mapRef = useRef(null);
+  const contactsRef = useRef(null);
+
+  const { venueDetails, isLoading } = useVenue(venueId);
+  const { venuesAmenities } = useLinkBar(DisplayedContentValue);
+  const theme = useTheme();
+  const isViewportLargerThanLg = useMediaQuery(theme.breakpoints.up("lg"));
+
+  const handleScroll = (ref) => {
+    if (ref?.current.offsetTop) {
+      window.scrollTo({
+        top: ref.current.offsetTop,
+        left: 0,
+        behavior: "smooth",
+      });
+    } else {
+      console.warn("Ref is null or undefined:", ref);
+    }
+  };
```

Comment:
This whole logic should be wrapped in the useVenue custom hook. When you have a component and there is a lot of logic you should wrap it in the hook named `use<Component name>`, for example a component `UserCard`should have a custom hook `useUserCard` with all the logic.

Reply:
i left some in the main file, hope thats ok:

 const { venuesAmenities } = useLinkBar(DisplayedContentValue);
  const contextIsError = useContext(Context)[0];

  const theme = useTheme();
  const isViewportLargerThanLg = useMediaQuery(theme.breakpoints.up("lg"));